sphinx/writers/latex.py

```diff
@@ -1991,14 +1992,15 @@ def visit_container(self, node):
             for id in self.pop_hyperlink_ids('code-block'):
                 ids += self.hypertarget(id, anchor=False)
             if node['ids']:
-                ids += self.hypertarget(node['ids'][0])
+                # suppress with anchor=False \phantomsection generation
+                ids += self.hypertarget(node['ids'][0], anchor=False)
             self.body.append('\n')
-            self.context.append(ids + '\n')
+            # context.pop will be done in visit_caption
+            self.context.append(ids)
```

Comment:
Pushing to and popping from context should be symmetrical.
It would be better to move these `ids` generation to `visit_caption`.